diff --git a/CMakeLists.txt b/CMakeLists.txt
index 966c5ccc..e23d1743 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -648,6 +648,9 @@ target_include_directories(rccl PRIVATE ${ROCM_SMI_INCLUDE_DIR})
 if(DEMANGLE_DIR)
   target_include_directories(rccl PRIVATE ${DEMANGLE_DIR})
 endif()
+if(DEFINED ENV{ROCMCORE_PATH})
+  target_include_directories(rccl PRIVATE "$ENV{ROCMCORE_PATH}/include")
+endif()

 ## Set RCCL compile definitions
 if(COLLTRACE)
